### ISSUE TITLE
to_chi() function giving surprising results

### ISSUE THREAD

**Issue by sodaH2O** [qutip/qutip#1873]
```
import numpy as np
import qutip as qt

m = np.array([[1,  2,  3,  4],
              [5,  6,  7,  8],
              [9,  10, 11, 12],
              [13, 14, 15, 16]])

sigmas = [qt.qeye(2), qt.sigmax(), qt.sigmay(), qt.sigmaz()]
s = 0
for i, sigma_i in zip(range(4), sigmas):
    for j, sigma_j in zip(range(4), sigmas):
        s += m[i, j] * qt.sprepost(sigma_i, sigma_j)
chi = qt.to_chi(s)
```

I expect `chi` to equal `m`, but the code gives `chi` as 4 times `m`, with negative entries on the 3rd row and column. Is there a mistake in the `to_chi` function?

**Comment by veronikakurth** [comment]
@pmenczel @nwlambert Are there any experts on the chi function? :) Or shall we close this issue as being not specific enough as for a bug report.

**Comment by nwlambert** [comment]
yeah there is something a bit odd here, or at least, to_chi() is using a convention that is not standard. 

About the factor of 4, the docs are a bit contradictory, they first imply that some kind of normalization is there, but then at the end there is a note that there is a factor of 4 needed to match some expected result from a normalized case.  probably needs cleaning up.

For the minus sign, I am less sure, but I guess that it comes from internally finding the chi() in the superoperator representation, so that it should be understood as a representation in terms of column-stacked superoperators (giving rise to sigma()^T = -sigmay()?).  I am not sure if that is an error (in the sense the minus sign should be compensated for at the end, to give expected result), or just an unusual convention, and the docs should be updated to clarify it?

**Comment by veronikakurth** [comment]
> yeah there is something a bit odd here, or at least, to_chi() is using a convention that is not standard.
> 
> About the factor of 4, the docs are a bit contradictory, they first imply that some kind of normalization is there, but then at the end there is a note that there is a factor of 4 needed to match some expected result from a normalized case. probably needs cleaning up.
> 
> For the minus sign, I am less sure, but I guess that it comes from internally finding the chi() in the superoperator representation, so that it should be understood as a representation in terms of column-stacked superoperators (giving rise to sigma()^T = -sigmay()?). I am not sure if that is an error (in the sense the minus sign should be compensated for at the end, to give expected result), or just an unusual convention, and the docs should be updated to clarify it?

Thank you very much for your feedback, Neill!
Also, issue #2259 might be related.

**Comment by qilesu** [comment]
Hi all, thank you for your responses

I think the minus sign is a bug rather than an unusual convention.

My best guess at where the bug originates from is the function `core.superop_reps._superpauli_basis`, whose output is used by `core.superop_reps._choi_to_chi`, which `to_chi` invokes. The output of `_superpauli_basis(nq=1)` currently is
```
Quantum object: dims=[[[2], [2]], [[2], [2]]], shape=(4, 4), type='super', dtype=CSR, isherm=False
Qobj data =
[[ 1.-0.j  0.+0.j  0.+0.j  1.-0.j]
 [ 0.+0.j  1.-0.j  0.-1.j  0.+0.j]
 [ 0.+0.j  1.-0.j  0.+1.j  0.+0.j]
 [ 1.-0.j  0.+0.j  0.+0.j -1.-0.j]]
```
I don't understand the CSR implementation, so I referenced the previous version of this function `_pauli_basis`, which seems to produce the same output for `nq=1`. Its seems the intention is to build the output matrix `B` by populating each of its columns with the column-stacked version of '_SINGLE_QUBIT_PAULI_BASIS = (identity(2), sigmax(), sigmay(), sigmaz())':
```
for idx, op in enumerate(starmap(tensor, product(_SINGLE_QUBIT_PAULI_BASIS, repeat=nq))):
    B[:, idx] = operator_to_vector(op).dag().data.todense()
```
For unknown reasons, `.dag()` was applied, which caused the complex conjugate to be inappropriately applied to all the entries, leading to the column for sigmay to pick up a minus sign. 

I think this behavior is not expected by other qutip functions because currently, the Pauli transfer matrix is computed using matrix `B` in  `core.superop_reps.to_superpauli` like so:
```
def to_superpauli(q_oper):
    sqobj = to_super(q_oper)
    ...
    out = B.dag() @ sqobj @ B
    ...
```
This transformation makes sense only if the columns of B are the column-stacked versions of the usual single-qubit Pauli basis. The minus sign error propagates to the Y row and Y column of the Pauli transfer matrix. The Pauli transfer matrix is conventionally defined using the basis I, X, Y, Z without any minus signs, as can be seen in the axis labels in `visualization.hinton`, which plots the output of `to_superpauli`. Incidentally, the author of `visualization.hinton` took the transpose of the data against the usual convention for plotting matrices, which I suspect is a mistaken attempt to counteract this bug.


The same `B` is used by `core.superop_reps._choi_to_chi`:
```
def _choi_to_chi(q_oper):
    ...
    return Qobj(_data.matmul(_data.matmul(B.adjoint(), q_oper.data), B),
    ...
```
If we reverse the sign of the third column of `core.superop_reps._superpauli_basis`, then the signs in the output of `to_chi` will be fixed.

**Comment by qilesu** [comment]
...though I am not sure how to make this work for `nq > 1`

**Comment by nwlambert** [comment]
@cgranade sorry to ping you on this, just wondered if you remembered much about this function?

**Comment by matti-sarjala** [comment]
A useful way to split this issue may be to separate the two effects into independent checks:

1. **Normalization / factor-of-4 convention**
   - Is `to_chi()` intended to return coefficients in an unnormalized Pauli basis or a normalized Pauli basis?
   - A documentation fix could explicitly show the expected scaling for the identity channel or for a simple Pauli channel.

2. **Y-sign / column-stacking convention**
   - This seems independent of the overall normalization.
   - A minimal test could check the image of a channel built from a single `sprepost(sigmay(), sigmay())` term and verify whether the resulting chi/PTM entry appears in the expected Y-basis position with the expected sign.

This might make the issue easier to resolve as either:
- a pure documentation clarification for the normalization convention,
- a sign-convention bug in the super-Pauli basis construction,
- or two separate follow-up issues / PRs.

I would avoid combining the normalization cleanup and the possible Y-sign fix into one PR unless the intended convention is first made explicit.

**Comment by michaelpasek** [comment]
@veronikakurth I would like to work on this issue. Could you please assign it to me? Thank you!

**Comment by veronikakurth** [comment]
> [@veronikakurth](https://github.com/veronikakurth) I would like to work on this issue. Could you please assign it to me? Thank you!

Please go ahead!
